Metalama.Framework.Engine/Validation/ReferenceIndexWalker.cs

```diff
@@ -75,18 +54,9 @@ public override void DefaultVisit( SyntaxNode node )
 
     public override void VisitIdentifierName( IdentifierNameSyntax node )
     {
-        this.ValidateNodeAndChildren( node, ReferenceKinds.Default );
-    }
-
-    public override void VisitGenericName( GenericNameSyntax node )
-    {
-        this.ValidateNodeWithoutChildren( node, ReferenceKinds.Default, node.Identifier );
-        this.Visit( node.TypeArgumentList );
-    }
+        var symbol = this._semanticModel!.GetSymbolInfo( node ).Symbol;
```

Comment:
Asking for SymbolInfo is very expensive. I think that avoiding it would be our main future performance goal (the whole method body is analyzed when you first ask for any node in the method). 

Is it possible to prefilter the references and ask for SymbolInfo only when needed? For example, it's certain "ClassA.Foo()" does not refer to "ClassA.Bar", so if I have a reference constraint only on Bar, I know I don't have to call GetSymbolInfo on that node. Probably for the future though.

Reply:
No this is not possible.